datafusion/optimizer/src/unwrap_cast_in_comparison.rs

```diff
@@ -653,4 +655,196 @@ mod tests {
     fn null_decimal(precision: u8, scale: u8) -> Expr {
         lit(ScalarValue::Decimal128(None, precision, scale))
     }
+
+    #[test]
+    fn test_try_cast_to_type_nulls() {
+        // test values that can be cast to/from all integer types
```

Comment:
It is a bit confusing that this description is the same as the one below, so might make sense to actually emphasize this is only testing null values for integer types.

Reply:
> It is a bit confusing that this description is the same as the one below, so might make sense to actually emphasize this is only testing null values for integer types.

agree, from the test case i got your mean.
You want to test that the `from value` is `Null`.

Reply:
This is a copy/pasted comment 🤦  -- I will fix it in the next PR. Sorry for the confusion and nice spot

Reply:
fixed in https://github.com/apache/arrow-datafusion/pull/4148